examples/smoke-co-alarm-app/telink/src/SmokeCoAlarmManager.cpp

```diff
@@ -0,0 +1,62 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "SmokeCoAlarmManager.h"
+#include "AppConfig.h"
+#include <platform/CHIPDeviceLayer.h>
+
+using namespace chip;
+using namespace ::chip::app::Clusters::SmokeCoAlarm;
+using namespace ::chip::DeviceLayer;
+
+LOG_MODULE_DECLARE(COsensor, CONFIG_CHIP_APP_LOG_LEVEL);
+
+SmokeCoAlarmManager SmokeCoAlarmManager::sAlarm;
+
+CHIP_ERROR SmokeCoAlarmManager::Init()
+{
+    ExpressedStateEnum currentExpressedState = ExpressedStateEnum::kNormal;
+    // read current ExpressedState on endpoint one
+    chip::DeviceLayer::PlatformMgr().LockChipStack();
+    SmokeCoAlarmServer::Instance().GetExpressedState(1, currentExpressedState);
+    chip::DeviceLayer::PlatformMgr().UnlockChipStack();
+
+    mExpressedState = currentExpressedState;
```

Comment:
Shouldn't the expressed state just always be provided by the app?  Asking the _cluster_ for that information is very strange.